### ISSUE TITLE
[Logic Bug] A Sepcial SQL Brings  Errors ( SELECT <column> FROM <table> WHERE <column> BETWEEN SYMMETRIC <column> AND (<decimal>::MONEY) )

### ISSUE THREAD

**Issue by qwebug** [heterodb/pg-strom#648]
#### Describe:

&#x20;Using SELECT \<column> FROM \<table> WHERE \<column> BETWEEN SYMMETRIC \<column> AND (\<decimal>::MONEY) brings different results, when turn on and off pg\_strom.enabled.

#### ***Sql:***

```sql
CREATE TABLE t1(c0 money , c1 int4range);
CREATE TABLE t2(LIKE t1);
INSERT INTO t1(c0, c1) VALUES(CAST(0.5 AS MONEY), ('[0,1]'::int4range));
INSERT INTO t2(c1, c0) VALUES('[0,1]'::int4range, (0.7)::MONEY);
CREATE SCHEMA extensions;
CREATE EXTENSION pg_strom WITH SCHEMA extensions;
set pg_strom.enabled=on;
SELECT t1.c0 FROM t2, t1 WHERE (t2.c0) BETWEEN SYMMETRIC (t1.c0) AND ((0.9)::MONEY);
```

#### ***Result:***

```bash
 c0 
----
(0 rows)
```

#### ***Sql:***

```sql
set pg_strom.enabled=off;
SELECT t1.c0 FROM t2, t1 WHERE (t2.c0) BETWEEN SYMMETRIC (t1.c0) AND ((0.9)::MONEY);
```

#### ***Result:***

```bash

  c0   
-------
 $0.50
(1 row)
```

### Environment:

Pg-strom Version: commit b959fef3547bb9050d8159bb2d8cd4aa4dc918d5

PostgreSQL Version: 15.3 on x86\_64-pc-linux-gnu, compiled by gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-18), 64-bit

CUDA Version: 12.0.1

NVIDIA Driver Version: 535.86.05

**Comment by kaigai** [comment]
It was wrong cast between currency and numeric. Fixed at `1f9fad1749fe22bce33d7a3a30b85ec550081b41`.
Thanks,

**Comment by qwebug** [comment]
This problem was not fixed at latest commit fb87671957502820db18d679be48b67418070a2e.
It still brings different results, when turn on and off pg_strom.enabled.

**Comment by kaigai** [comment]
It was fixed at `b2868103c807f9ed23df6c5e1ed55441f77a1fc7`

Thanks for your report.
